src/python/pants/backend/python/tasks/resolve_requirements.py

```diff
@@ -24,7 +25,28 @@ def product_types(cls):
   def prepare(cls, options, round_manager):
     round_manager.require_data(PythonInterpreter)
 
+  @classmethod
+  def register_options(cls, register):
+    super(ResolveRequirements, cls).register_options(register)
+
+    register('--force', type=bool, default=False, fingerprint=True,
+             help='Always run requirements resolution, even if all the target roots are '
+                  'python_binary()s.')
+
   def execute(self):
+    # TODO: PythonBinaryCreate resolves its own requirements separately, more sensitively to
+    # platform constraints. No-oping here is a bit of a hack and will cause repls on python_binary()
+    # targets to fail.
+    if all(PythonBinaryCreate.is_binary(t) for t in self.context.target_roots):
+      if self.get_options().force:
```

Comment:
I recommend inverting the conditionals for an early return. Will allow you to get rid of the `else` clause and decrease the indentation for the next line.